packages/flutter/lib/src/material/menu_bar.dart

```diff
@@ -0,0 +1,3190 @@
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/scheduler.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+
+import 'button_style.dart';
+import 'color_scheme.dart';
+import 'divider.dart';
+import 'icons.dart';
+import 'material.dart';
+import 'material_localizations.dart';
+import 'material_state.dart';
+import 'menu_bar_theme.dart';
+import 'text_button.dart';
+import 'text_button_theme.dart';
+import 'theme.dart';
+import 'theme_data.dart';
+
+// The time after a menu item is opened via hover during which clicks on a child
+// menu item with a submenu will be ignored by the menu item. This is so that
+// users aren't surprised by hovering over a main menu to open it, and then
+// quickly moving to a submenu and clicking on it to open it. Without this
+// delay, it often just feels like the menu opened briefly and closed when
+// clicked on, when in reality they opened it via the hover before they ever
+// clicked on it.
+const Duration _kMenuHoverClickBanDelay = Duration(milliseconds: 500);
+
+// The default size of the arrow that indicates that a menu has a submenu.
+const double _kDefaultSubmenuIconSize = 24.0;
+
+// The default spacing between the the leading icon, label, trailing icon, and
+// shortcut label in a _MenuBarItemLabel.
+const double _kLabelItemDefaultSpacing = 18.0;
+
+// The minimum spacing between the the leading icon, label, trailing icon, and
+// shortcut label in a _MenuBarItemLabel.
+const double _kLabelItemMinSpacing = 4.0;
+
+// The minimum horizontal spacing on the outside of the top level menu.
+const double _kTopLevelMenuHorizontalMinPadding = 4.0;
+
+const Map<ShortcutActivator, Intent> _kMenuTraversalShortcuts = <ShortcutActivator, Intent>{
+  SingleActivator(LogicalKeyboardKey.enter): ActivateIntent(),
+  SingleActivator(LogicalKeyboardKey.numpadEnter): ActivateIntent(),
+  SingleActivator(LogicalKeyboardKey.space): ActivateIntent(),
+  SingleActivator(LogicalKeyboardKey.gameButtonA): ActivateIntent(),
+  SingleActivator(LogicalKeyboardKey.escape): DismissIntent(),
+  SingleActivator(LogicalKeyboardKey.tab): NextFocusIntent(),
+  SingleActivator(LogicalKeyboardKey.tab, shift: true): PreviousFocusIntent(),
+  SingleActivator(LogicalKeyboardKey.arrowDown): DirectionalFocusIntent(TraversalDirection.down),
+  SingleActivator(LogicalKeyboardKey.arrowUp): DirectionalFocusIntent(TraversalDirection.up),
+  SingleActivator(LogicalKeyboardKey.arrowLeft): DirectionalFocusIntent(TraversalDirection.left),
+  SingleActivator(LogicalKeyboardKey.arrowRight): DirectionalFocusIntent(TraversalDirection.right),
+};
+
+/// A menu bar with cascading child menus.
+///
+/// This is a Material Design menu bar that resides above the main body of an
```

Comment:
Can't it really go anywhere at all? Maybe replace "that resides" with "that typically resides".